### DISCUSSION TITLE
`assemblyof(MyAssemblyName)` keyword/operator

### DISCUSSION THREAD

**Discussion by jzabroski** [dotnet/csharplang#6229] [Language Ideas] [answered]
I commonly write stuff like this:

```c#
var assembliesToScan = new[]
            {
                typeof(Global).Assembly, // Website project.
                typeof(ModelMappingService).Assembly // Web project.
            };
```

But this is kind of annoying because it assumes the presence of a type.

Similar to the `typeof` operator that returns a Type, `assemblyof` would return an Assembly.

I realize this is not the highest priority idea ever suggested, but it would make the code slightly more readable and is unlikely to draw a maelstrom of hate from people saying we're ruining the language by making it too complex.

**Comment by foxesknow** [comment]
> Similar to the typeof operator that returns a Type, assemblyof would return an Assembly.

What would you pass to `assemblyof` to help it identify another assembly?

**Reply by jzabroski** [reply]
That could be complicated. :-) Right now, the `[assembly: Attribute]` operator takes a raw string.

The reason it could be complicated is that MSBuild/.NET Common Project System attributes determine the final assembly name.

I suppose if `assemblyof` were added, you could write:

```c#
[assembly: InternalsVisibleToAttribute(assemblyof(My.Assembly.Name))]
```

However, that would only provide [unsigned assembly names](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute?view=net-6.0#examples). I personally don't care for signed assembly names, since I think merkle trees would be a better way to describe the lineage of an assembly. Plus, it is incomprehensible when reading a PublicKeyToken hash in a peer review whether it is correct. For any solution-wide usages, a C# Source Generator that shadows InternalsVisibleTo could potentially provide a public key token.

**Comment by HaloFour** [comment, marked as answer]
See: https://github.com/dotnet/csharplang/discussions/2715

**Reply by jzabroski** [reply]
Thanks, there is no way to 'close' a discussion, so I'll mark this as an answer.  I did search for related ideas, but I guess I should have just searched for "assemblyof" rather than my complex search phrase `¯\_(ツ)_/¯`
